cmd/minikube/cmd/docker-env.go

```diff
@@ -245,6 +245,7 @@ var dockerEnvCmd = &cobra.Command{
 		shl := shell.ForceShell
 		if shl == "" {
 			shl, err = shell.Detect()
+			shl = strings.Trim("powershell.exe", ".exe")
```

Comment:
The result of `strings.Trim("powershell.exe", ".exe")` is always `powershell`. Here we effectively just set shl = "powershell" and ignore the result of `shell.Detect()`. Not sure it's that we really need here.

Reply:
@hetong07 this seems like a mistake to me, I wonder if u meant to do something else ? simmilar to ilya I suggest we fix the shell.Detect() logic so it does the right thing

Reply:
@medyagh and @ilya-zuyev Sorry I mistakenly upload the experimental changes here, and have fixed it in subsequent CL, do I need to make a --force push to overwrite this commit?

Reply:
@medyagh and @ilya-zuyev do a check and find my branch has divert from the minikube HEAD, so I will close issue and create another PR to solve this problem.